linera-chain/src/manager.rs

```diff
@@ -500,7 +500,7 @@ where
     pub fn create_final_vote(
         &mut self,
         validated: ValidatedBlockCertificate,
-        key_pair: Option<&SigningKey>,
+        key_pair: Option<&AuthorityPrivateKey>,
```

Comment:
Don't we actually want key pairs instead of secret keys here?

Reply:
We can, but it was the PrivateKey-only before this as well and public can be derived.

Reply:
OK, something for a separate PR as well then.

I was just wondering whether there's any computational overhead if we don't have the public key available here.

Reply:
Probably there is - especially that for secp256k1 keys it will go through FFI. We can change that later.